sentry-android-core/src/main/java/io/sentry/android/core/ActivityLifecycleIntegration.java

```diff
@@ -0,0 +1,198 @@
+  private final @NotNull WeakHashMap<Activity, ITransaction> activitiesWithOngoingTransactions =
+      new WeakHashMap<>();
+
+  public ActivityLifecycleIntegration(final @NotNull Application application) {
+    this.application = Objects.requireNonNull(application, "Application is required");
+  }
+
+  @Override
+  public void register(final @NotNull IHub hub, final @NotNull SentryOptions options) {
+    this.options =
+        Objects.requireNonNull(
+            (options instanceof SentryAndroidOptions) ? (SentryAndroidOptions) options : null,
+            "SentryAndroidOptions is required");
+
+    this.hub = Objects.requireNonNull(hub, "Hub is required");
+
+    this.options
+        .getLogger()
+        .log(
+            SentryLevel.DEBUG,
+            "ActivityLifecycleIntegration enabled: %s",
+            this.options.isEnableActivityLifecycleBreadcrumbs());
+
+    performanceEnabled = isPerformanceEnabled(this.options);
+
+    if (this.options.isEnableActivityLifecycleBreadcrumbs() || performanceEnabled) {
+      application.registerActivityLifecycleCallbacks(this);
+      options.getLogger().log(SentryLevel.DEBUG, "ActivityLifecycleIntegration installed.");
+    }
+  }
+
+  private boolean isPerformanceEnabled(final @NotNull SentryAndroidOptions options) {
+    return ((options.getTracesSampleRate() != null || options.getTracesSampler() != null)
+        && options.isEnableAutoActivityLifecycleTracing());
+  }
+
+  @Override
+  public void close() throws IOException {
+    application.unregisterActivityLifecycleCallbacks(this);
+
+    options.getLogger().log(SentryLevel.DEBUG, "ActivityLifecycleIntegration removed.");
+  }
+
+  private void addBreadcrumb(final @NonNull Activity activity, final @NotNull String state) {
+    if (options.isEnableActivityLifecycleBreadcrumbs()) {
+      final Breadcrumb breadcrumb = new Breadcrumb();
+      breadcrumb.setType("navigation");
+      breadcrumb.setData("state", state);
+      breadcrumb.setData("screen", getActivityName(activity));
+      breadcrumb.setCategory("ui.lifecycle");
+      breadcrumb.setLevel(SentryLevel.INFO);
+      hub.addBreadcrumb(breadcrumb);
+    }
+  }
+
+  private @NotNull String getActivityName(final @NonNull Activity activity) {
+    return activity.getClass().getSimpleName();
+  }
+
+  private void startTracing(final @NonNull Activity activity) {
+    if (performanceEnabled && !isRunningTransaction(activity)) {
+      final ITransaction transaction =
+          hub.startTransaction(getActivityName(activity), "navigation");
+
+      // lets bind to the scope so other integrations can pick it up
+      hub.configureScope(
+          scope -> {
+            scope.withTransaction(
+                scopeTransaction -> {
+                  // we we'd not like to overwrite existent transactions bound to the Scope
+                  // manually.
+                  if (scopeTransaction == null) {
+                    scope.setTransaction(transaction);
+                  }
```

Comment:
worth a log.debug("Activity transaction won't be bound since {scope.transaction.name} is alread ythere?

Reply:
good point